### ISSUE TITLE
Isn't Robot still useful?

### ISSUE THREAD

**Issue by akitchen** [jeffh/Robot#9]
Just saw https://github.com/jeffh/Robot/commit/6541b962a4b02f8df21dcb3231236f781a924668

Isn't this still useful for certain testing strategies?

**Comment by jeffh** [comment]
Hey @akitchen,

Correct me if I'm wrong, but I just don't see a decent future in any direction of Robot. Apple's new [UITesting](https://medium.com/@larcus94/ui-testing-with-xcode-7-221d16bad276) covers what Robot does.

Robot also is still limited in many ways - most obvious is it only runs inside Cedar Spec Suites IIRC. And its api needs a full rewrite to support Swift. The only thing in Robot that might be worth salvaging is the `RBTimeLapse` file.

The existing deployed versions of Robot are still available on cocoapods, but I don't really think I can recommend it for Xcode 7+.

If you think it's worth restoring, I'm ok with that. But it's not my primary focus at this point.

**Comment by joemasilotti** [comment]
As of Beta 2, UI Testing is still pretty limited. For example, there's no notion of "waiting." You can't wait for a view to appear or disappear. You can't even wait for `x` seconds! I did a breakdown of some of the limitations [towards the end of this post](http://masilotti.com/ui-testing-xcode-7).

**Comment by jeffh** [comment]
Hey @joemasilotti,

I haven't done much UI Testing TBH, but this seems to work for me when waiting for a given amount of time:

``` swift
// Application changes text of button after 2 seconds.
NSRunLoop.mainRunLoop().runUntilDate(NSDate().dateByAddingTimeInterval(3))
let button = XCUIApplication().buttons["Hello world"]
XCTAssertTrue(button.exists)
```

**Comment by joemasilotti** [comment]
Good call, @jeffh. I put together some of these into a pod, [JAMTestHelper](http://github.com/joemasilotti/JAMTestHelper), last weekend.
